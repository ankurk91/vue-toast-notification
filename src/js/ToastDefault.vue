<template>
  <div
    role="alert"
    :class="['v-toast__item', `v-toast__item--${type}`]"
  >
    <div class="v-toast__icon"></div>
    <p class="v-toast__text" v-html="message"></p>
  </div>
</template>

<script>
import {defineComponent} from 'vue';

export default defineComponent({
  name: 'Toast',
  props: {
    message: {
      type: String,
      required: true
    },
    type: {
      type: String,
      default: 'success'
    },
  },
})
</script>
