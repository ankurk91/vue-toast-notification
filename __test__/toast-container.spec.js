import {mount} from '@vue/test-utils';
import ToastContainer from '../src/js/ToastContainer.vue';
import ToastDefault from '../src/js/ToastDefault.vue';
import { h } from 'vue';

describe('ToastContainer component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(ToastContainer, {
      propsData: {
        duration: 5000
      },
      global: {
        stubs: {
          transition: false
        },
      },
      slots: {
        default: h(ToastDefault, {
          type: 'success',
          message: 'Hello world'
        })
      }
    });
  });

  test('renders when active', async () => {
    await wrapper.vm.$nextTick();
    expect(wrapper.vm.$el).toMatchSnapshot()
  });

  test('renders passed component', async () => {
    expect(wrapper.html()).toContain('v-toast__item')
  })

});
