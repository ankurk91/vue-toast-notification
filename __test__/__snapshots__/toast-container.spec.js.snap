// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToastContainer component renders when active 1`] = `
<div
  class="v-toast__container v-toast__container--bottom-right v-enter-from v-toast--fade-in-up"
  style=""
>
  
  <div
    class="v-toast__item v-toast__item--success"
    role="alert"
  >
    <div
      class="v-toast__icon"
    />
    <p
      class="v-toast__text"
    >
      Hello world
    </p>
  </div>
  
</div>
`;
